File: pkgcloud/firstapp-getting-started_4.js
var pkgcloud = require('pkgcloud');
var yaml = require('yamljs');

var config = yaml.load( 'clouds.yaml' );
var openstack = pkgcloud.compute.createClient({
    provider:     "openstack",
    username:     config[ "clouds" ][ "internapNYJ" ][ "auth" ].username,
    password:     config[ "clouds" ][ "internapNYJ" ][ "auth" ].password,
    authUrl:      config[ "clouds" ][ "internapNYJ" ][ "authUrl" ],
    region:       config[ "clouds" ][ "internapNYJ" ][ "region_name" ]
});

openstack.getImages(function (err, images)
{
    if (err) { console.log(err); return; }

    // Pick an image based on Ubuntu 14.04
    var image = null;
    for( var currentImageIndex = 0, len = images.length; currentImageIndex  < len; currentImageIndex++ )
    {
        if( images[ currentImageIndex  ].id != null &&
            images[ currentImageIndex  ].id == '3c76334f-9644-4666-ac3c-fa090f175655' )
            image = images[ currentImageIndex  ];
    }
    console.log( 'Selected image : ' +  JSON.stringify( image ) );

    openstack.getFlavors(function (err, flavors)
    {
        if (err) { console.log(err); return; }

        // Pick the smallest instance flavor
        var flavor = null;
        for( var currentFlavorIndex = 0, len = flavors.length; currentFlavorIndex < len; currentFlavorIndex++ ) {
            if( flavors[ currentFlavorIndex ].id != null &&
                flavors[ currentFlavorIndex ].id == 'A1.1' )
                flavor = flavors[ currentFlavorIndex ];
        }
        console.log( 'Selected flavor : ' + JSON.stringify( flavor ) );
        console.log( "Done! Congrats" );
    });
});

File: libcloud/firstapp-getting-started_4.py
#! /usr/bin/env bash

from libcloud.compute.types import Provider
from libcloud.compute.providers import get_driver
import os_client_config

cloud_config = os_client_config.OpenStackConfig().get_one_cloud(
    'internapNYJ', region_name='nyj01')

provider = get_driver(Provider.OPENSTACK)
conn = provider(cloud_config.config['auth']['username'],
                cloud_config.config['auth']['password'],
                ex_force_auth_url=cloud_config.config['auth']['auth_url'],
                ex_force_auth_version='2.0_password',
                ex_tenant_name=cloud_config.config['auth']['project_name'],
                ex_force_service_region=cloud_config.region)

# step-4
image_id = '3c76334f-9644-4666-ac3c-fa090f175655'
image = conn.get_image(image_id)
print( 'Selected image : ' + image)

# step-5
flavor_id = 'A1.1'
flavor = conn.ex_get_size(flavor_id)
print('Selected flavor : ' + flavor)

print( 'Done! Congrats!')
